spec/controllers/form_drafts_controller_spec.rb

```diff
@@ -232,6 +232,53 @@
     end
   end
 
+  describe 'POST #create' do
+    let :submit do
+      post :create, form_draft: { name: 'test' }, commit: @commit
+    end
+    context 'not staff' do
+      before :each do
+        when_current_user_is :not_staff
+      end
+      it 'does not allow access' do
+        expect_any_instance_of(FormDraft).not_to receive :create
+        submit
+        expect(response).to have_http_status :unauthorized
+      end
+    end
+    context 'staff' do
+      before :each do
+        when_current_user_is :staff
+      end
+      context 'commit is Save changes and continue editing' do
+        before :each do
+          @commit = 'Save changes and continue editing'
+        end
+        it 'creates a form with the changes given' do
+          expect { submit }.to change { Form.count }.by 1
+          expect(assigns[:draft].form.name).to eql 'test'
+        end
+        it 'redirects to the edit path for the draft' do
+          submit
+          expect(response).to redirect_to edit_form_draft_path(assigns[:draft])
```

Comment:
I don't think it's a good idea to use part of the result of the code you're testing as the expectation for how that code should behave.